shelf/spiders/books.py

```diff
@@ -0,0 +1,54 @@
+import re
+
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> dict:
+        for book_card in response.css("article.product_pod"):
+            yield response.follow(
+                book_card.css("a::attr(href)").get(),
+                callback=self.parse_book_detail
+            )
+
+        if next_href := response.css("li.next a::attr(href)").get():
+            yield response.follow(next_href, callback=self.parse)
+
+    @staticmethod
+    def parse_book_detail(response: Response) -> dict:
+        product_detail = dict(
+            zip(
+                response.css("table tr th::text").getall(),
+                response.css("table tr td::text").getall()
+            )
+        )
+        rating = {
+            "One": 1,
+            "Two": 2,
+            "Three": 3,
+            "Four": 4,
+            "Five": 5
+        }
```

Comment:
make it a constant and move to the top of the file so that it's easier to find and edit it